guides/plugins/plugins/framework/rule/add-custom-rules.md

```diff
@@ -0,0 +1,200 @@
+
+### Creating rule in PHP
+
+First of all we need a new Rule class, in this example we will name it `LunarEclipseRule`. It will be placed in the directory `<plugin root>/src/Core/Rule`.
+Our new class has to extend from the abstract class `Shopware\Core\Framework\Rule\Rule`. Below you can find an example implementation.
+
+{% code title="<plugin root>/src/Core/Rule/LunarEclipseRule.php" %}
+```php
+<?php declare(strict_types=1);
+
+namespace Swag\BasicExample\Core\Rule;
+
+use Shopware\Core\Framework\Rule\Rule;
+use Shopware\Core\Framework\Rule\RuleScope;
+use Symfony\Component\Validator\Constraints\Type;
+
+class LunarEclipseRule extends Rule
+{
+    /**
+     * @var bool
+     */
+    protected $isLunarEclipse;
+
+    public function __construct()
+    {
+        parent::__construct();
+
+        // Will be overwritten at runtime. Reflects the expected value.
+        $this->isLunarEclipse = false;
+    }
+
+    public function getName(): string
+    {
+        return 'lunar_eclipse';
+    }
+
+    public function match(RuleScope $scope): bool
+    {
+        // Not implemented in this example
+        $isCurrentlyLunarEclipse = $this->isCurrentlyLunarEclipse();
+
+        // Checks if the shop owner set the rule to "Lunar eclipse => Yes"
+        if ($this->isLunarEclipse) {
+            // Shop administrator wants the rule to match if there's currently a lunar eclipse.
+            return $isCurrentlyLunarEclipse;
+        }
+
+        // Shop administrator wants the rule to match if there's currently NOT a lunar eclipse.
+        return !$isCurrentlyLunarEclipse;
+    }
+
+    public function getConstraints(): array
+    {
+        return [
+            'isLunarEclipse' => [ new Type('bool') ]
+        ];
+    }
+}
+```
+{% endcode %}
+
+As you can see, several methods are already implemented:
+
+- `__constructor`: This only defines the default expected value. This is overwritten at runtime with the actual value, that the shop owner set in the administration.
+- `getName`: Returns a unique technical name for your rule.
+- `match`: This checks whether the rule applies. Accordingly, a boolean is returned whether the rule applies or not.
+- `getConstraints`: This method returns an array of the possible fields and its types. You could also return the `NotBlank` class here, to require this field.
+
+After we've created our rule class, we have to register it in our `services.xml` and tag it as `shopware.rule.definition`.
+
+### Active rules
+
+You can access all active rules by using the `getRuleIds` method of the context.
+
+```php
+$context->getRuleIds();
+```
+
+### Showing rule in the administration
+
+Now we want to implement our new rule in the administration that we can manage it. To achieve this, we have to call the `addCondition` method of the [RuleConditionService](https://github.com/shopware/platform/blob/v6.3.4.1/src/Administration/Resources/app/administration/src/app/service/rule-condition.service.js), by decorating this service.
```

Comment:
```suggestion
Now we want to implement our new rule in the administration so that we can manage it. To achieve this, we have to call the `addCondition` method of the [RuleConditionService](https://github.com/shopware/platform/blob/v6.3.4.1/src/Administration/Resources/app/administration/src/app/service/rule-condition.service.js), by decorating this service.
```